includes/template-library/manager.php

```diff
@@ -764,4 +756,21 @@ private function ensure_args( array $required_args, array $specified_args ) {
 
 		return true;
 	}
+
+	private function is_allowed_to_read_template( array $args ): bool {
+		if ( 'remote' === $args['source'] ) {
+			return true;
+		}
+
+		if ( null === $this->wordpress_adapter ) {
+			$this->set_wordpress_adapter( new WordPress_Adapter() );
+		}
+
+		$post_id = intval( $args['template_id'] );
+		$post_status = $this->wordpress_adapter->get_post_status( $post_id );
+		$is_private_or_non_published = ( 'private' === $post_status && ! $this->wordpress_adapter->current_user_can( 'read_private_posts', $post_id ) ) || ( 'publish' !== $post_status );
```

Comment:
```suggestion
		$is_private_or_non_published = ( 'private' === $post_status && ! $this->wordpress_adapter->current_user_can( 'read_private_posts', $post_id ) ) || ( 'publish' !== $post_status && ! $this->wordpress_adapter->current_user_can( 'edit_post', $post_id ) );
```
I think the previous check from the revert was better. This part that you added `( 'publish' !== $post_status )` was a false fix, the issue was with the `source`.

WDYT? CC @mykytamurzin @mugurelLupu